### PR TITLE
Adds extra confirmation to payinvoice and sendpayment, can be bypassed by force.

### LINKED ISSUE #1455: lncli: add extra confirmation dialogue to payinvoice+sendpayment
Currently there's no easy way for a user to verify the integrity of a payment request they're about to pay, unless one first uses the `decodepayreq` method _each_ time before sending on the command line. It would be nice if this became the default, as since many users are new to the system, it's easy for a site to trick a user into paying a higher invoice. To avoid this, we should add a default mode where we _always_ display the invoice to the user for confirmation before attempting to pay the invoice. 

# Steps To Completion 

- [ ] Modify `payinvoice` and `sendpayment` to _always_ first call `DecodePayReq` with the payreq. The response of this should be displayed to the users w/ a confirmation dialogue w.r.t if we should proceed or not. 

- [ ] Add a `--force` (or `-f`) flag to both of these commands to skip the user verification

### PR DESCRIPTION
**Pull request by rawtxapp**
Resolves #1455.

Also looks like gofmt wasn't run on this file since the last commit that checked-in closedchannels, so I ran it too.

### CODE REVIEW

File: cmd/lncli/commands.go [hunk 1]
```diff
@@ -1915,6 +1974,13 @@ func payInvoice(ctx *cli.Context) error {
 		return err
 	}
 
+	if !ctx.IsSet("force") {
+		err = confirmPayreq(ctx)
```
Comment (wpaulino) [inline cmd/lncli/commands.go]: This will end up parsing the payment request twice, since it's already done above. I would much rather move the parsing of the payment request outside of `confirmPayreq`. This way we can avoid passing in the `cli.Context` and can just pass in the payment request string.

File: cmd/lncli/commands.go [hunk 2]
```diff
@@ -1729,6 +1733,51 @@ func retrieveFeeLimit(ctx *cli.Context) (*lnrpc.FeeLimit, error) {
...
+	client, cleanUp := getClient(ctx)
+	defer cleanUp()
+
+	req := &lnrpc.PayReqString{PayReq: payReq}
+	resp, err := client.DecodePayReq(ctxb, req)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Description: %v\n", resp.GetDescription())
```
Comment (wpaulino) [inline cmd/lncli/commands.go]: We can use `promptForConfirmation` here and avoid all the code below this.

File: cmd/lncli/commands.go [hunk 3]
```diff
@@ -1729,6 +1733,51 @@ func retrieveFeeLimit(ctx *cli.Context) (*lnrpc.FeeLimit, error) {
 	return nil, nil
 }
 
+func confirmPayreq(ctx *cli.Context) error {
```
Comment (wpaulino) [inline cmd/lncli/commands.go]: nit: s/confirmPayreq/confirmPayReq/

File: cmd/lncli/commands.go [hunk 4]
```diff
@@ -1733,19 +1733,7 @@ func retrieveFeeLimit(ctx *cli.Context) (*lnrpc.FeeLimit, error) {
...
-	switch {
-	case ctx.IsSet("pay_req"):
-		payReq = ctx.String("pay_req")
-	case args.Present():
-		payReq = args.First()
-	default:
-		return fmt.Errorf("pay_req argument missing")
-	}
-
+func confirmPayReq(ctx *cli.Context, payReq string) error {
```
Comment (wpaulino) [inline cmd/lncli/commands.go]: We can actually pass in the `lnrpc.LightningClient` rather than the `cli.Context` like we do in `executeChannelClose` :)

Comment (rawtxapp) [reply cmd/lncli/commands.go]: Sure, I did the change, but it means that the client creation is duplicated at both function calls.

Comment (wpaulino) [reply cmd/lncli/commands.go]: Yep, so then we can also just do the same thing with `sendPaymentRequest` and then we can have the client definition at the top of `payInvoice` like all the other commands.

Comment (rawtxapp) [reply cmd/lncli/commands.go]: Sounds good, done.

**Comment by wpaulino** [comment]
Awesome! LGTM. Could you restructure your commits to better follow the [contribution guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#ModelGitCommitMessages)?

**Comment by rawtxapp** [comment]
Done, renamed commits like "lncli:...", less than 50 chars descriptions and also squashed the 1-2 liner changes that I made during the review.

File: cmd/lncli/commands.go [hunk 5]
```diff
@@ -1915,6 +1952,13 @@ func payInvoice(ctx *cli.Context) error {
 		return err
 	}
 
+	if !ctx.IsSet("force") {
+		err = confirmPayReq(ctx, payReq)
+		if err != nil {
+			return err
+	}
```
Comment (halseth) [inline cmd/lncli/commands.go]: indentation

File: cmd/lncli/commands.go [hunk 6]
```diff
@@ -1702,7 +1702,7 @@ var sendPaymentCommand = cli.Command{
 		cli.BoolFlag{
 			Name:  "force, f",
 			Usage: "will skip payment request confirmation",
-	},
+		},
```
Comment (halseth) [inline cmd/lncli/commands.go]: the formatting shouldn't be a separate commit

**Comment by rawtxapp** [comment]
Thanks for the review. I've squashed into a single commit.

**Comment by cfromknecht** [comment]
Thanks @rawtxapp, this is a feature i've been wanting for a while! PR LGTM, will need a rebase before merging. May not be merged immediately, so I'd hold off on that until we figure out a merge order

**Comment by rawtxapp** [comment]
Thanks for the review, let me know whenever you guys want to merge it and I'll rebase.

**Comment by cfromknecht** [comment]
Awesome, sounds good :)

File: cmd/lncli/commands.go [hunk 7]
```diff
@@ -1750,13 +1777,19 @@ func sendPayment(ctx *cli.Context) error {
 	// If a payment request was provided, we can exit early since all of the
 	// details of the payment are encoded within the request.
 	if ctx.IsSet("pay_req") {
+		if !ctx.IsSet("force") {
```
Comment (halseth) [inline cmd/lncli/commands.go]: I think this should be `!ctx.Bool("force")`, as now it would skip the confirmation even when setting `--force=0`.

**Comment by rawtxapp** [comment]
Changed to using ctx.Bool and rebased on master.
